src/browser_action/on-each-interaction.js

```diff
@@ -11,31 +11,71 @@
  limitations under the License.
 */
 
+import {INPThresholds} from './web-vitals.js';
+
 /**
  * @param {Function} callback
  */
 export function onEachInteraction(callback) {
-  const valueToRating = (score) => score <= 200 ? 'good' : score <= 500 ? 'needs-improvement' : 'poor';
+  const valueToRating = (score) => score <= INPThresholds[0] ? 'good' : score <= INPThresholds[1] ? 'needs-improvement' : 'poor';
 
   const observer = new PerformanceObserver((list) => {
+    const entries = list.getEntries();
     const interactions = {};
 
-    for (const entry of list.getEntries().filter((entry) => entry.interactionId)) {
+    const getSelector = (node, maxLen) => {
+      let sel = '';
+
+      try {
+        while (node && node.nodeType !== 9) {
+          const el = node;
+          const part = el.id
+            ? '#' + el.id
+            : getName(el) +
+              (el.classList &&
+              el.classList.value &&
+              el.classList.value.trim() &&
+              el.classList.value.trim().length
+                ? '.' + el.classList.value.trim().replace(/\s+/g, '.')
+                : '');
+          if (sel.length + part.length > (maxLen || 100) - 1) return sel || part;
+          sel = sel ? part + '>' + sel : part;
+          if (el.id) break;
+          node = el.parentNode;
+        }
+      } catch (err) {
+        // Do nothing...
+      }
+      return sel;
+    };
+
+    for (const entry of entries.filter((entry) => entry.interactionId)) {
       interactions[entry.interactionId] = interactions[entry.interactionId] || [];
       interactions[entry.interactionId].push(entry);
     }
-
     // Will report as a single interaction even if parts are in separate frames.
     // Consider splitting by animation frame.
     for (const interaction of Object.values(interactions)) {
       const entry = interaction.reduce((prev, curr) => prev.duration >= curr.duration ? prev : curr);
       const value = entry.duration;
+      const interactionId = entry.interactionId;
+
+      // Filter down LoAFs to ones that intersected any event startTime and any processingEnd
+      const longAnimationFrameEntries = getIntersectingLoAFs(entry.startTime, entry.startTime + entry.value)
+
+      let firstEntryWithTarget = interaction[0].target;
+      firstEntryWithTarget = null;
+      if (!firstEntryWithTarget) {
+        firstEntryWithTarget = Object.values(interactions).flat().find(entry => entry.interactionId === interactionId && entry.target)?.target;
+      }
 
       callback({
         attribution: {
-          eventEntry: entry,
-          eventTime: entry.startTime,
-          eventType: entry.name,
+          interactionTarget: getSelector(firstEntryWithTarget),
+          interactionTime: entry.startTime,
+          interactionType: entry.name.startsWith('key') ? 'keyboard' : 'pointer',
+          longAnimationFrameEntries: longAnimationFrameEntries,
+          processedEventEntries: entries,
```

Comment:
`processedEventEntries` will mean something different here compared to the INP attribution, which could be confusing. Maybe it's better to just omit this for interaction logging?

Reply:
Done.